### PR TITLE
Create hook update to update resources without group but dataset pare…

### PR DESCRIPTION
**Pull request by jacintocapote**
…nt has groups

### CODE REVIEW

File: modules/dkan_dataset_groups/dkan_dataset_groups.install [hunk 1]
```diff
@@ -22,4 +22,49 @@ function dkan_dataset_groups_install() {
...
+ * Update resources without groups with the same group from parent dataset.
+ */
+function dkan_dataset_groups_update_7000(&$sandbox) {
+  if (!isset($sandbox['progress'])) {
+    $sandbox['progress'] = 0;
+    $sandbox['max'] = db_query("SELECT COUNT(DISTINCT fdr.field_dataset_ref_target_id) FROM {node} n LEFT JOIN {field_data_field_dataset_ref} fdr ON fdr.entity_id=n.nid LEFT JOIN {og_membership} og ON n.nid=og.etid WHERE n.type='resource' AND og.etid IS NULL AND fdr.field_dataset_ref_target_id IN (SELECT etid FROM {og_membership} WHERE etid IN (SELECT field_dataset_ref_target_id FROM {field_data_field_dataset_ref}))")->fetchField();
+  }
+
+  //Get all resources witout group but the parent dataset has groups
+  $result = db_query("SELECT fdr.field_dataset_ref_target_id FROM {node} n LEFT JOIN {field_data_field_dataset_ref fdr} ON fdr.entity_id=n.nid LEFT JOIN {og_membership} og ON n.nid=og.etid WHERE n.type='resource' AND og.etid IS NULL AND fdr.field_dataset_ref_target_id IN (SELECT etid FROM {og_membership} WHERE etid IN (SELECT field_dataset_ref_target_id FROM {field_data_field_dataset_ref})) LIMIT 0,10");
```
Comment (dafeder) [inline modules/dkan_dataset_groups/dkan_dataset_groups.install:52]: We are wondering if this query will catch and correctly update resources that belong to two different datasets.

File: modules/dkan_dataset_groups/dkan_dataset_groups.install [hunk 2]
```diff
@@ -22,4 +22,49 @@ function dkan_dataset_groups_install() {
...
+
+  foreach ($result as $item) {
+    //Simulate a empty original
```
Comment (dafeder) [inline modules/dkan_dataset_groups/dkan_dataset_groups.install:55]: Same question as above - does this work correctly with resources that belong to more than one dataset? If not, we may want to refactor to use different logic than dkan_dataset_sync_groups().

File: modules/dkan_dataset_groups/dkan_dataset_groups.install [hunk 3]
```diff
@@ -22,4 +22,49 @@ function dkan_dataset_groups_install() {
...
+  // update. Without this dependecy dkan_dataset_groups_update_7000
+  // runs before throwing an entity metadata wrapper error.
+  $dependencies['dkan_dataset_groups'][7000] = array(
+    'dkan_datastore' => 7001,
+  );
+  return $dependencies;
+}
+
+/**
+ * Update resources without groups with the same group from parent dataset.
```
Comment (dafeder) [inline modules/dkan_dataset_groups/dkan_dataset_groups.install:43]: I think we need a bit more documentation to explain what this is for.

File: modules/dkan_dataset_groups/dkan_dataset_groups.install [hunk 4]
```diff
@@ -22,4 +22,49 @@ function dkan_dataset_groups_install() {
 function dkan_dataset_groups_disable() {
   //Remove field groups from the index.
   features_revert(array('dkan_sitewide_search_db' => array('search_api_index', 'search_api_server')));
-}
\ No newline at end of file
+}
+
+/**
```
Comment (dafeder) [inline modules/dkan_dataset_groups/dkan_dataset_groups.install:27]: We should actually be doing all updates in dkan.install, not individual module install files. Move this and the datastore ones to dkan.install (eliminating need for dependencies and extra PRs).

**Review by dafeder — CHANGES_REQUESTED** [review]
A couple questions about the update hook @jacintocapote

**Comment by ghost** [comment]
This code has been moved to 'dkan.install' here https://github.com/NuCivic/dkan/pull/1491. Closing this.

**Pull request closed without merging**